url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/Trading_Engine/myDatabase.java

```diff
@@ -0,0 +1,571 @@
+						if(i == 2){
+							newHead += " , " + "millisecond";
+						}
+					}
+					//System.out.println(tableQuery);
+					s.execute("create table " + dbName + "(" + tableQuery + ")");
+					s.execute("create table old_trade_list(" + tableQuery + ")");
+					insertAllDatabase(connection,br,newHead,dbName,overviewResult);
+				} else {
+					System.out.println("Error: file content not csv format");
+				}
+			}else {
+				System.out.println("Error: file is empty");
+			}
+			br.close();
+			s.close();
+		}catch (Exception e) {
+			System.out.println("In insertAll:  " + e);
+		}
+		return overviewResult;
+	}
+	public void deleteAllTables(){
+		try {
+			Statement statement = connection.createStatement();
+			DatabaseMetaData dbm = connection.getMetaData();
+			String[] types = {"TABLE"};
+			ResultSet allTables = dbm.getTables(null,null,"%",types);
+			if(allTables!=null){
+				while(allTables.next()){
+					String table = allTables.getString("TABLE_NAME");
+					statement.addBatch("drop table " + table);
+					System.out.println("delete table: " + table);
+				};
+				statement.executeBatch();
+			}
+			allTables.close();
+			statement.close();
+		}catch(Exception e) {
+			System.out.println("in deleteAllTables: " + e);
+		}
+	}
+	public static void insertQuery(PreparedStatement pstmt, String[] insertElement){
+		try {
+			int lengthOfArray = insertElement.length;
+			//System.out.println(insertElement.length);
+			pstmt.setString(1, insertElement[0]);
+
+			String tempDate = insertElement[1].substring(0, 4) + "-" 
+					+ insertElement[1].substring(4, 6) + "-" + insertElement[1].substring(6, 8);
+			pstmt.setDate(2, Date.valueOf(tempDate));
+			pstmt.setTime(3, Time.valueOf(insertElement[2].substring(0, 8)));
+			pstmt.setInt(4, Integer.parseInt(insertElement[2].substring(9)));
+			pstmt.setString(5, insertElement[3]);
+			if(insertElement[4].isEmpty()){
+				pstmt.setNull(6, java.sql.Types.DOUBLE);
+			}else {
+				pstmt.setDouble(6, Double.parseDouble(insertElement[4]));
+			}
+			if(insertElement[5].isEmpty()){
+				pstmt.setNull(7, java.sql.Types.INTEGER);
+			}else {
+				pstmt.setInt(7, Integer.parseInt(insertElement[5]));
+			}
+			if(insertElement[6].isEmpty()){
+				pstmt.setNull(8, java.sql.Types.INTEGER);
+			}else {
+				pstmt.setInt(8, Integer.parseInt(insertElement[6]));
+			}
+			if(lengthOfArray > 7){
+				if(insertElement[7].isEmpty()){
+					pstmt.setNull(9, java.sql.Types.FLOAT);
+				}else {
+					pstmt.setFloat(9, Float.parseFloat(insertElement[7]));
+				}
+			}
+			if(lengthOfArray > 8){
+				if(insertElement[8].isEmpty()){
+					pstmt.setNull(10, java.sql.Types.VARCHAR);
+				}else {
+					pstmt.setString(10, insertElement[8]);
+				}
+			}
+			if(lengthOfArray > 9){
+				if(insertElement[9].isEmpty()){
+					pstmt.setNull(11, java.sql.Types.INTEGER);
+				}else{
+					pstmt.setInt(11, Integer.parseInt(insertElement[9]));
+				}
+			}else{
+				pstmt.setNull(11, java.sql.Types.INTEGER);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 235:243, 244:252, 253:261, 290:298, 299:307, 308:316.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.